src/FTPC.jl

```diff
@@ -502,24 +472,18 @@ Establish connection to FTP server.
 returns ctxt::ConnContext
 """ ->
 function ftp_connect(options::RequestOptions=RequestOptions())
-    if options.blocking
-        ctxt = false
-        try
-            ctxt = setup_easy_handle(options)
-
-            # ping the server
-            resp = ftp_command(ctxt, "LIST")
+    ctxt = nothing
+    try
+        ctxt = setup_easy_handle(options)
```

Comment:
Do the setup prior to the `try`: http://docs.julialang.org/en/release-0.4/manual/control-flow/#finally-clauses